compiler/codegen/OMRPeephole.hpp

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#ifndef OMR_PEEPHOLE_INCL
+#define OMR_PEEPHOLE_INCL
+
+/*
+ * The following #define and typedef must appear before any #includes in this file
+ */
+#ifndef OMR_PEEPHOLE_CONNECTOR
+#define OMR_PEEPHOLE_CONNECTOR
+namespace OMR { class Peephole; }
+namespace OMR { typedef OMR::Peephole PeepholeConnector; }
+#endif
+
+#include "env/TRMemory.hpp"
+#include "infra/Flags.hpp"
+#include "infra/Annotations.hpp"
+
+namespace TR { class CodeGenerator; }
+namespace TR { class Instruction; }
+namespace TR { class Peephole; }
+
+namespace OMR
+{
+
+class OMR_EXTENSIBLE Peephole
+   {
+   public:
+
+   TR_ALLOC(TR_Memory::UnknownType)
+
+   Peephole(TR::Compilation* comp);
+   
+   TR::Peephole* self();
+
+   TR::CodeGenerator* cg() const;
+   TR::Compilation* comp() const;
+
+   public:
+
+   virtual bool perform();
+   virtual bool performOnInstruction(TR::Instruction* cursor);
+
+   private:
+
+   TR::Compilation* _comp;
+   TR::CodeGenerator* _cg;
```

Comment:
You're right, technically we can just grab the compilation from codegen. I was just modeling how other extensible classes looked since some appear to have both. I'd definitely want a second opinion on this (@0xdaryl?), but for the record I'm happy to make the change if we all agree. We should likely make a similar change in other extensible classes in the codegen where this is prevalent. I can open an issue for that if we agree this is the direction we want to take.

Reply:
@0xdaryl thoughts on the above?

Reply:
Do you think we should do the work now (just in the Peephole changes) or have a blanked issue to cover all other classes and do them all at once at a later point?

Reply:
Not urgent in my mind.  This can be cleaned up later.  I only commented because my opinion was asked. :-)